### PR TITLE
minor change plus code cleanup and refactoring

### PR DESCRIPTION
**Pull request by doxiao**
Changes in this PR:
1) let operator continue, instead of fail, when it hits Unauthorized error on creating NamespaceWatchingStopped event in the domain's namespace. This only happens when a namespace is not managed by the operator any more, and the permissions that are required to create an event in that namespace have been removed; the operator already logs this error.
2) clean up some classes based on warnings that IntelliJ points out.
3) minor refactoring in DomainRecheck as a followup to a review comment to PR#2180. 

Integration test run on Jenkins (completed with 2 known failures): https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/4102/.

### CODE REVIEW

File: operator/src/main/java/oracle/kubernetes/operator/DomainStatusUpdater.java [hunk 1]
```diff
@@ -529,17 +530,22 @@ private boolean stillHasPodPendingRestart(DomainStatus status) {
```
Comment (russgold) [inline operator/src/main/java/oracle/kubernetes/operator/DomainStatusUpdater.java:530]: How can you be certain that the pod will never be null?

There are a number of steps here, and I'd be inclined to use a stream to resolve them, such as:

```
      private boolean stillHasPodPendingRestart(DomainStatus status) {
        return status.getServers().stream()
              .map(this::getServerPod)
              .map(this::getLabels)
              .anyMatch(m -> m.containsKey(LabelConstants.MII_UPDATED_RESTART_REQUIRED_LABEL));
      }

      private V1Pod getServerPod(ServerStatus serverStatus) {
        return getInfo().getServerPod(serverStatus.getServerName());
      }
```

**Review by russgold — CHANGES_REQUESTED** [review]
Thanks for doing some cleanup. There are a couple more things I think you should do here.

File: operator/src/main/java/oracle/kubernetes/operator/DomainStatusUpdater.java (hunk 1 above)
Comment (doxiao) [reply operator/src/main/java/oracle/kubernetes/operator/DomainStatusUpdater.java:530]: If the pod is null getLabels() will return an empty map, and the check for containsKey will be false.
But I like your suggestion

**Review by ankedia — APPROVED** [review]
LGTM

**Review by russgold — APPROVED** [review]

**Pull request merged**